### DISCUSSION TITLE
Is it possible to load a (global) configuration that does not care about the filetype?

### DISCUSSION THREAD

**Discussion by ametis70** [mfussenegger/nvim-dap#950] [Q&A]
I have a configuration in my launch.json file that starts a debugging session for a server, and it does not care about the file I'm currently editing, so I was wondering if it's possible to use something like 

```lua
require('dap.ext.vscode').load_launchjs(nil, {['pwa-node'] = { '*' } }) 
```

to make the command accesible from `require(dap).continue()` from any filetype.

Thanks in advance!

**Comment by mfussenegger** [comment]
It's currently not possible

**Comment by AlephNought0** [comment]
Hello, will such feature be ever implemented? I just started using neovim and I love it. Your plugin is also nice and makes debugging applications really easy on neovim. However, needing to go to a specific filetype every time I want to debug an application is really limiting and slows everything down for me. I tried looking around but I don't see anything about this feature being implemented so I am asking here.

**Reply by mfussenegger** [reply]
It actually was implemented, see https://github.com/mfussenegger/nvim-dap/pull/1237

You can add a custom provider that either goes through all filetypes, or looks for configurations some other place.
Configurations in a local `.vscode/launch.json` now also always show up, independent of the filetype.

**Comment by sleeptightAnsiC** [comment]
Hey @AlephNought0, @ametis70 

After using the plugins for quite some time, I've found a good solution for this.
Instead of using `dap.continue`, use `dap.run`.
You can define multiple wrappers for said function that dynamically assemble configuration that you need.
For example, this one runs the script file that you're currently using with appropriate interpreter:
```
dap.run({
	type = vim.bo.filetype,
	request = "launch",
	name = "Launch script with custom arguments (adhoc)",
	program = "${file}",
	args = vim.split(vim.fn.expand(t.args), "\n"),
})
```
I have a lot of recipes like this one in my config. I usually define a special user command for each.

This becomes super powerful and flexible when you get used to it.
You can probably even easily parse already existing dap.configurations and run debuggers based on them.
For more examples see https://github.com/mfussenegger/nvim-dap/wiki/Cookbook

With this, I don't think my comment above comment is relevant, I will delete it.
